lib/winapi/timezoneapi.c

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: MIT
+
+// SPDX-FileCopyrightText: 2023 Ryan Wendland
+
+#include <timezoneapi.h>
+#include <winbase.h>
+#include <assert.h>
+#include <xboxkrnl/xboxkrnl.h>
+
+typedef struct {
+    UCHAR Month;     // 1-12
+    UCHAR Day;       // 1 = 1st occurrence of DayOfWeek up to 5 (5th or last)
+    UCHAR DayOfWeek; // 0 = Sunday to 6 = Saturday
+    UCHAR Hour;
+} XBOX_TZ_STRUCT;
+
+// Determine the day of the week given a year, month and day
+// https://en.wikipedia.org/wiki/Determination_of_the_day_of_the_week#Methods_in_computer_code
+static UCHAR GetDayofWeek(SHORT y, UCHAR m, UCHAR d) {
```

Comment:
nit: `GetDaysinMonth` functions use `year`, `month` spelled out; should be more consistent?

Reply:
Done